extensions/web/09_file.js

```diff
@@ -483,25 +464,35 @@
      * underlying bytes.
      *
      * @param {number} start
-     * @param {number} len
+     * @param {number} end
      * @returns {BlobReference}
      */
-    slice(start, len) {
-      const id = core.opSync("op_blob_slice_part", this.#id, { start, len });
-      return new BlobReference(id);
+    slice(start, end) {
+      const size = end - start;
+      const id = core.opSync("op_blob_slice_part", this._id, { start, len: size });
+      return new BlobReference(id, size);
     }
 
-    // TODO(lucacasonato): this should be a stream!
     /**
      * Read the entire contents of the reference blob.
-     *
-     * @returns {Promise<Uint8Array>}
+     * @returns {AsyncGenerator<Uint8Array>}
      */
-    read() {
-      return core.opAsync("op_blob_read_part", this.#id);
+    async *stream() {
+      let position = 0;
+      const end = this.size;
+      while (position !== end) {
+        const size = Math.min(end - position, POOL_SIZE);
+        const chunk = this.slice(position, position + size);
+        position += chunk.size;
+        yield core.opAsync("op_blob_read_part", chunk._id);
+      }
     }
   }
 
+  // TODO(jimmywarting): Maybe remove BlobDataItem?
```

Comment:
Yeah, I think it can go now that we have `BlobReference`. It is essentially the same thing, just one uses a dedicated API.